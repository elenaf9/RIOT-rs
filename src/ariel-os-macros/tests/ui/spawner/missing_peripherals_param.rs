#![no_main]
#![feature(used_with_arg)]

// As the macro will fail, this import will not get used
#[allow(unused_imports)]
use ariel_os::Spawner;

// FAIL: the `peripherals` parameter is required in this case
#[ariel_os::spawner(autostart)]
fn main(_spawner: Spawner, _peripherals: Peripherals) {}

struct Peripherals;
