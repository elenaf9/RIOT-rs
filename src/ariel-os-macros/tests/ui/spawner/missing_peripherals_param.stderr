error: `peripherals` macro parameter missing here ...
 --> tests/ui/spawner/missing_peripherals_param.rs:9:1
  |
9 | #[ariel_os::spawner(autostart)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the attribute macro `ariel_os::spawner` (in Nightly builds, run with -Z macro-backtrace for more info)

error: ... because this function has a second parameter
  --> tests/ui/spawner/missing_peripherals_param.rs:10:9
   |
10 | fn main(_spawner: Spawner, _peripherals: Peripherals) {}
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
